qemu/tests/migration_with_dst_problem.py

```diff
@@ -438,8 +438,8 @@ def test(self):
             self.copy_timeout = int(params.get("copy_timeout", "1024"))
 
             self.copy_block_size = int(utils_misc.normalize_data_size(
-                params.get("copy_block_size", "100M"), "M"))
-            self.disk_size = "%sM" % int(self.copy_block_size * 1.4)
+               params.get("copy_block_size", "100M"), "M").split(".")[0])
+            self.disk_size = int(self.copy_block_size * 1.4)
```

Comment:
From your commit message header: `Fix ValueError: invalid literal for int() with base 10`, I think we can only focus on it.
And for line 442, 455 and 631~632, I guess it maybe another issue about the design flaw.